File: asgmt03/src/library/Book.java
package library;

import java.util.Iterator;

class Book
	extends Item
{
    private String author;
    private int pageNum;

    Book(String title, String newAuthor, int nPages, String... keywords) {
        super(title, keywords);
        author = newAuthor;
        pageNum = nPages;
    }

    public String getAuthor() {
        return author;
    }

    public String toString() {
        StringBuilder bookString = new StringBuilder();
        Iterator<String> it = keywords.iterator();

        bookString.append("-Book- \n").append("author:  ").append(author).append('\n');
        bookString.append("# pages:  ").append(pageNum).append('\n');
        bookString.append("title:    ").append(title).append('\n');

        bookString.append("keywords: ");
        while (it.hasNext()) {
            String s = it.next();
            bookString.append(s);
            if (it.hasNext()) {
                bookString.append(',');
            }
        }
        bookString.append('\n').append('\n');
        return bookString.toString();
    }

}

File: asgmt04/Book.cpp
#include "memoryleakdetect.h"		// needs to be first #include in the .cpp file
#include "Book.h"

Book::Book(void)
{
}

Book::Book(string newTitle, string newAuthor, int newNPages): Item(newTitle, newAuthor)
{
	int nPages = newNPages;
}


Book::~Book(void)
{
}

